Compile/src/Producers/PrefiringWeightProducer.cc

```diff
@@ -0,0 +1,159 @@
+            double eta_gam = photon.p4.Eta();
+            if (pt_gam < 20.)
+                continue;
+            if (fabs(eta_gam) < 2.)
+                continue;
+            if (fabs(eta_gam) > 3.)
+                continue;
+            double prefiringprob_gam = getPrefiringRate(eta_gam, pt_gam, h_prefmap_photon, fluct);
+            nonPrefiringProba[fluct] *= (1. - prefiringprob_gam);
+        }
+        // applying the prefiring maps to jets in the affected regions.
+        for (unsigned int iJet = 0; iJet < event.m_tjets->size();  ++iJet)
+        {
+            KLV jet;
+            jet.p4 = event.m_tjets[iJet].p4;
+            double pt_jet = jet.p4.Pt();
+            double eta_jet = jet.p4.eta();
+            if (pt_jet < 20.)
+                continue;
+            if (fabs(eta_jet) < 2.)
+                continue;
+            if (fabs(eta_jet) > 3.)
+                continue;
+            //Loop over photons to remove overlap
+            double nonprefiringprobfromoverlappingphotons = 1.;
+            for (unsigned int iph = 0; iph<product.m_pfPhotons.size(); ++iph)
+            {
+                KLV photon;
+                photon.p4 = product.m_pfPhotons[iph]->p4;
+                double pt_gam = photon.p4.Pt();
+                double eta_gam = photon.p4.eta();
+                if (pt_gam < 20.)
+                    continue;
+                if (fabs(eta_gam) < 2.)
+                    continue;
+                if (fabs(eta_gam) > 3.)
+                    continue;
+                double dR = ROOT::Math::VectorUtil::DeltaR(photon.p4, jet.p4);
+                if (dR > 0.4)
+                    continue;
+                double prefiringprob_gam = getPrefiringRate(eta_gam, pt_gam, h_prefmap_photon, fluct);
+                nonprefiringprobfromoverlappingphotons *= (1. - prefiringprob_gam);
+
+            }
+            double nonprefiringprobfromoverlappingjet = 1. - getPrefiringRate(eta_jet, pt_jet, h_prefmap_jet, fluct);
+            if (nonprefiringprobfromoverlappingphotons == 1.)
+            {
+                nonPrefiringProba[fluct] *= nonprefiringprobfromoverlappingjet;
+            }
+            //If overlapping photons have a non prefiring rate larger than the jet, then replace these weights by the jet one
+            else if (nonprefiringprobfromoverlappingphotons > nonprefiringprobfromoverlappingjet)
+            {
+                if (nonprefiringprobfromoverlappingphotons != 0.)
+                {
+                    nonPrefiringProba[fluct] *= nonprefiringprobfromoverlappingjet / nonprefiringprobfromoverlappingphotons;
+                } else
+                {
+                    nonPrefiringProba[fluct] = 0.;
+                }
+            }
+            //Last case: if overlapping photons have a non prefiring rate smaller than the jet, don't consider the jet in the event weight, and do nothing.
+        }
+    }
+    product.m_weights["prefiringWeight"] =  1./nonPrefiringProba[0];
```

Comment:
Hi Matthias,
 you shouldn't add these Weights to `m_weights` because these are all multiplied to get the `weight` branch in the NTuples. There should be also a `m_optionalWeight` object available, which is not directly related to the `weight` quantity.

Reply:
HI Maximilian,
that's a good point. I will change it. However, for data no `EventWeightProducer` is running. Therefore, weights are not multiplied to `weight`. The `weight` in the ntuple is still `1`.

Reply:
Unfortunatly I merged that PR into master with the new commit.